app/controllers/save_pages_controller.rb

```diff
@@ -0,0 +1,28 @@
+class SavePagesController < ApplicationController
+  include AccountConcern
+
+  before_action :set_no_cache_headers
+  before_action :authenticate
+
+  def create
+    redirect_to GdsApi.account_api.save_page(
+      page_path: params[:page_path],
+      govuk_account_session: account_session_header,
+    ).to_h["saved_page"]["page_path"]
+  rescue GdsApi::HTTPUnauthorized
+    @account_session_header = nil
+    authenticate
+  end
+
+private
+
+  def authenticate
+    unless logged_in?
+      redirect_to new_govuk_session_path(
```

Comment:
This will have to be `redirect_with_ga` to pass along the `_ga` param if present.